_posts/2018-10-22-release-1-5.md

```diff
@@ -0,0 +1,199 @@
+checking tool to compliment pmemcheck.
+
+To keep this relatively short, here I'll just briefly describe each item, and
+the bigger ones will get their own posts describing the details of the change.
+
+## Reliability, Availability and Serviceability (RAS)
+
+NVDIMMs are storage and as such have to deal with on-media errors and hardware
+failures. But they are also memory, which means that an application can
+encounter poisoned memory pages at runtime, and what's more, those poisoned
+pages persist across restarts and need to be handled manually.
+
+To account for that, we've worked on two main RAS features:
+
+### Unsafe (dirty) shutdown count
+
+This is used to detect if an ADR, feature on which persistent memory programming
+model relies on to avoid having to flush the memory controller caches, failure
+has occurred while a pool has been opened. Based on this information, the
+library can determine whether the pool could have been corrupted or not.
+
+### Bad blocks support
+
+On Linux, if application maps a persistent memory resident file with
+a bad block, everything will work correctly up until the bad, poisoned, page
+is accessed for the first time in the current instance of the application. That
+first page fault will cause a SIGBUS and the process will most likely be
+terminated.
+
+To prevent this cycle from happening over and over again, PMDK will now detect
+if the files that the pool is composed of have any badblocks, and if so, it will
+refuse to open it. This allows the user to handle the failure gracefully.
+
+In addition to that, we implemented recovery of bad blocks from poolset
+replicas.
+
+## pmemreorder
+
+Almost since the beginning of this project, we planned on implementing a tool
+which could exhaustively check consistency for all possible combination of
+persistent memory stores based on runtime binary instrumentation.
+This would give us, and our users, confirmation that the algorithms inside of
+our applications are in fact correct and fail-safe atomic.
+
+We already had two takes on this problem, with the second one being almost good
+enough for generic use. And so, after almost two years, we have finally decided
+that it's time to revisit the topic and cleanup the old codebase so that it can
+be released alongside pmemcheck.
+
+Stay tuned for a blog post about pmemreorder. In the meantime, see its
+[man page](http://pmem.io/pmdk/manpages/linux/master/pmreorder/pmreorder.1.html).
+
+## Flexible memcpy
+
+Right from the start, libpmem shipped with custom-made memcpy/memset
+implementations that made use of non-temporal stores (stores that do not go
+through the CPU cache) in a deterministic fashion, which we leveraged to
+optimize PMDK algorithms. But we've discovered optimization opportunities
+that required even more control over the behavior of these functions, and so
+we decided to add new variants of libpmem primitives (flush, memcpy, ...) that
+take flags which control how the functions behave.
```

Comment:
This is not clear enough... It's supposed to talk about "Flexible memcpy", but then talks about "flush" primitve.